src/rgw/rgw_rados.cc

```diff
@@ -2378,6 +2379,22 @@ int RGWRados::Bucket::update_bucket_id(const string& new_bucket_id)
 }
 
 
+static inline std::string after_delim(std::string_view delim)
```

Comment:
careful here, i think you're reintroducing the listing performance bug we just fixed

Reply:
Thanks, I did not look closely enough. I thought I was just moving this before the comment, but some old code got dragged in as well.